client/lib/protect-form/index.tsx

```diff
@@ -16,10 +12,13 @@ import { Subtract } from 'utility-types';
  * Module variables
  */
 const debug = debugModule( 'calypso:protect-form' );
-let formsChanged = [];
+
+type ComponentMarkedWithFormChanges = Component;
```

Comment:
After a session exploring this with @dmsnell, it seems like this is a satisfactory solution.

- This is local so no overly permissive types are being spread
- It's been extremely difficult to get the type right on the higher order component.
- As far as the implementation, things are being added to and removed from a list (`any`).
- As far as application logic, we only expect to work with `Component`s, and indeed everywhere this is used matches that type.
- Aliasing as `ComponentMarkedWithFormChanges` lets us communicate a bit more about what's going on and would hopefully discourage misuse.